### ISSUE TITLE
Packaged WinUI 3 app can only p/invoke local DLLs on ARM when run as Admin

### ISSUE THREAD

**Issue by wbokkers** [microsoft/microsoft-ui-xaml#9335]
### Describe the bug

We have issues with our packaged x64 WinUI3 app when running on ARM64. This app runs without any issues on x64.

P/Invoke to DLLs in the application folder does not work, unless the app is started as Administrator.

The app runs with full trust (as it is a WinUI 3 packaged app). 

I tried to target as arm64 as well, but this gave me even more issues. Running as administrator did not help solve the issue in this case.

### Steps to reproduce the bug

1. Install Xelion on an ARM device from this installer: https://appinstaller.xelion.com/winapp/v8n/Xelion-Package.appinstaller
2. Run Xelion
3. Enter credentials (if Microsoft reaches out, I can supply test credentials)
4. The app crashes on trying to p/invoke into the local dlls (sipdll.dll)

To successfully run the app, run Xelion as Administrator. 

### Expected behavior

The app should be able to p/invoke DLLs from the installation folder without the need for Admin privileges.

### NuGet package version

WinUI 3 - Windows App SDK 1.4.5: 1.4.240411001

### Windows version

Windows 11 (22H2): Build 22621

**Comment by wbokkers** [comment]
I have more information. 
When running the app as a normal (non Admin) user, the  x64 DLL cannot load. I get the following exception:

System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (0x8007000B)

But as soon as the app runs as Admin, the DLL loads without any issue.

**Comment by DarranRowe** [comment]
Have you tried doing the same thing with an application compiled for ARM64? I appreciate that the library you are trying to load may not have an ARM64 equivalent, so having a smaller test would be useful. Also, have you checked the ACL on the DLL that you are trying to load? Just to be sure that Windows didn't somehow mangle the DACL when the package was installed.

**Comment by wbokkers** [comment]
@DarranRowe Yes, I did try to build for ARM64, but the DLLs remain x64, so this does not work either.
I did not check the ACL/DACL, I never do and I can't remember having issues with it in the 30 years I work on Windows. The installed package works just fine on x64 machines and I cannot imagine that the installation is somehow different on ARM-systems, but who knows. I will take a look. Thanks for the suggestion.
